tests/testthat/test-grapes-.-grapes.R

```diff
@@ -40,3 +40,21 @@ test_that("%.% works for nested input", {
   expect_identical(d %.% a %.% f %.% g, 3)
   expect_error(d %.% a %.% f %.% h, "h not found in d %.% a %.% f")
 })
+
+
+test_that("%.% works for generalized context", {
+
+  d <- list(a = \() 2, b = c(3, 4), e = list("f" = 5))
+
+  # Test function context
+  expect_identical(d$a(), 2)
+  expect_identical(d %.% a(), 2)
+
+  # Test vector context
+  expect_identical(d$b[1], 3)
+  expect_identical(d %.% b[1], 3)
+
+  # Test $ context
+  expect_identical(d$e$f, 5)
+  expect_identical(d %.% e$f, 5)
```

Comment:
Why not d %.% e %.% f ?

Reply:
That case is being tested in the test before this one.
This test is looking at "mixed" usage of operators and ensuring it works as expected